semantic-scope-graph/src/Control/Carrier/Sketch/Fresh.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | This carrier interprets the Sketch effect, keeping track of
+-- the current scope and in-progress graph internally.
+module Control.Carrier.Sketch.Fresh
+  ( SketchC (..)
+  , runSketch
+  , module Control.Effect.Sketch
+  ) where
+
+import           Control.Algebra
+import           Control.Carrier.Fresh.Strict
+import           Control.Carrier.State.Strict
+import           Control.Effect.Sketch
+import           Control.Monad.IO.Class
+import           Data.Bifunctor
+import           Data.Module
+import           Data.Name (Name)
+import qualified Data.Name
+import           Data.ScopeGraph (ScopeGraph)
+import qualified Data.ScopeGraph as ScopeGraph
+import           Data.Semilattice.Lower
+import           Source.Span
+import qualified System.Path as Path
+
+data Sketchbook address = Sketchbook
```

Comment:
I’m still confused about the `Sketch`/`Sketchbook` nomenclature; could you please 📝 the choice of name, perhaps in the module header?